integrationtests/agent/adoption_test.go

```diff
@@ -0,0 +1,387 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+)
+
+func init() {
+	resources["BundleDeploymentConfigMap"] = []v1alpha1.BundleResource{
+		{
+			Name: "configmap.yaml",
+			Content: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cm1
+data:
+  key: value
+`,
+			Encoding: "",
+		},
+	}
+}
+
+var _ = Describe("Adoption", Label("adopt"), func() {
+	var (
+		namespace string
+		env       *specEnv
+	)
+
+	createBundleDeployment := func(name string, takeOwnership bool) {
```

Comment:
I think that this test suite would be more readable if these helper functions were moved _after_ the actual test case implementations, and perhaps outside this `Describe` block.

Reply:
I thought about putting them into specEnv, but wasn't confident enough to do that. Do you have an opinion about that?